core/autoscaling.go

```diff
@@ -25,16 +25,17 @@ type autoScalingGroup struct {
 	config              AutoScalingConfig
 }
 
-func (a *autoScalingGroup) loadLaunchConfiguration() error {
+func (a *autoScalingGroup) loadLaunchConfiguration() (*launchConfiguration, error) {
 	//already done
 	if a.launchConfiguration != nil {
-		return nil
+		return a.launchConfiguration, nil
 	}
 
 	lcName := a.LaunchConfigurationName
 
 	if lcName == nil {
-		return errors.New("missing launch configuration")
```

Comment:
Correct me if I'm wrong, but this doesn't seem like an error. I think it would return an error for all ASGs using LaunchTemplates. I believe this only worked since we were ignoring any errors returned by `loadLaunchConfiguration`.

Reply:
I agree, good catch!